### PR TITLE
[Mux/Demux] accept flex tensor

### PR DESCRIPTION
**Pull request by jaeyun-jung**
add pad template to accept flex tensor, and update source to configure tensor info for static/flex tensor stream.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

File: tests/nnstreamer_sink/unittest_sink.cc [hunk 1]
```diff
@@ -95,7 +95,9 @@ typedef enum {
   TEST_TYPE_FLEX_TENSOR_1, /**< pipeline for flexible tensor (sink) */
   TEST_TYPE_FLEX_TENSOR_2, /**< pipeline for flexible tensor (converter, static multi-tensors to flex) */
   TEST_TYPE_FLEX_TENSOR_3, /**< pipeline for flexible tensor (converter, flex to static) */
-  TEST_TYPE_TENSORS, /**< pipeline for tensors with tensor_mux */
+  TEST_TYPE_TENSORS_MUX_1, /**< pipeline for tensors with tensor_mux */
+  TEST_TYPE_TENSORS_MUX_2, /**< pipeline for tensors with tensor_mux */
```
Comment (dongju-chae) [inline tests/nnstreamer_sink/unittest_sink.cc:99]: Could you add extra comments how this is different from `TEST_TYPE_TENSORS_MUX_1`

Comment (jaeyun-jung) [reply tests/nnstreamer_sink/unittest_sink.cc]: updated

**Review by dongju-chae — APPROVED** [review]
LGTM

File: tests/nnstreamer_sink/unittest_sink.cc [hunk 2]
```diff
@@ -1411,6 +1431,93 @@ TEST (tensorSinkTest, capsTensors)
...
+  EXPECT_TRUE (gst_tensors_config_validate (&g_test_data.tensors_config));
+  EXPECT_EQ (g_test_data.tensors_config.info.num_tensors, 1U);
+  EXPECT_EQ (g_test_data.tensors_config.info.info[0].dimension[0], 3U * 320 * 240);
+  EXPECT_EQ (g_test_data.tensors_config.rate_n, 30);
+  EXPECT_EQ (g_test_data.tensors_config.rate_d, 1);
+
+  EXPECT_FALSE (g_test_data.test_failed);
+  _free_test_data (option);
+}
+
```
Comment (myungjoo) [inline tests/nnstreamer_sink/unittest_sink.cc:1520]: Add more negative test cases (more than positive test cases), with _n postfixes.

**Review by myungjoo — COMMENTED** [review]
Question: what's the GST CAP of mux (flex-tensor, static-tensor)? It appears that there is no updates on negotiation mechanisms in this PR.

**Comment by jaeyun-jung** [comment]
> Question: what's the GST CAP of mux (flex-tensor, static-tensor)? It appears that there is no updates on negotiation mechanisms in this PR.

@myungjoo all type. (other/tensor tensors and tensors-flexible)
For tensor-mux, the common util functions have the negotiation mechanisms now.

**Review by myungjoo — APPROVED** [review]

**Pull request merged**